File: challenges/008/goodVsEvil.js
function goodVsEvil(good, evil){

    const goodValues = [1, 2, 3, 3, 4, 10]
    const evilValues = [1, 2, 2, 2, 3, 5, 10]

    const goodInt = good.split(" ").map(x => Number(x))
    const evilInt = evil.split(" ").map(x => Number(x))

    const goodArray = []
    const evilArray = []

    for (let i = 0; i < goodValues.length; i++) {
        goodArray.push(goodInt[i] * goodValues[i])
    }

    for (let i = 0; i < evilValues.length; i++) {
        evilArray.push(evilInt[i] * evilValues[i])
    }

    const goodScore = goodArray.reduce((accumulator, currentValue) => {
        return accumulator + currentValue;
    }, 0)


    const evilScore = evilArray.reduce((accumulator, currentValue) => {
        return accumulator + currentValue;
    }, 0)

    if (goodScore > evilScore) {
        return "Battle Result: Good triumphs over Evil"
    } else if (goodScore < evilScore ) {
        return "Battle Result: Evil eradicates all trace of Good"
    } else return "Battle Result: No victor on this battle field"
}

File: challenges/008/good_vs_evil.py
GOOD_WORTH = [1, 2, 3, 3, 4, 10]
EVIL_WORTH = [1, 2, 2, 2, 3, 5, 10]


def good_vs_evil(good, evil):

    good_list = good.split(" ")
    evil_list = evil.split(" ")

    good_values = []
    evil_values = []

    for i in range(len(GOOD_WORTH)):
        good_values.append(GOOD_WORTH[i] * int(good_list[i]))

    for i in range(len(EVIL_WORTH)):
        evil_values.append(EVIL_WORTH[i] * int(evil_list[i]))

    good_score = sum(good_values)
    evil_score = sum(evil_values)

    if good_score > evil_score:
        return "Battle Result: Good triumphs over Evil"
    elif good_score < evil_score:
        return "Battle Result: Evil eradicates all trace of Good"
    return "Battle Result: No victor on this battle field"


# Simpler, more pythonic version:
def good_vs_evil(good, evil):
    good_score = sum(w * int(c) for w, c in zip(GOOD_WORTH, good.split()))
    evil_score = sum(w * int(c) for w, c in zip(EVIL_WORTH, evil.split()))

    if good_score > evil_score:
        return "Battle Result: Good triumphs over Evil"
    elif evil_score > good_score:
        return "Battle Result: Evil eradicates all trace of Good"
    return "Battle Result: No victor on this battle field"